File: backend/src/main/java/com/UnCypher/models/dto/ChatMessage.java
package com.UnCypher.models.dto;

import lombok.Data;
import java.time.Instant;

@Data
public class ChatMessage {
    private String messageId; // UUID or ULID
    private String userId;     // user UUID
    private String role;       // "user" or "assistant"
    private String content;    // actual message text
    private Instant timestamp; // when the message was generated
}

File: frontend/src/models/ChatMessage.js
// Basic ChatMessage structure (documented for dev reference)
export function createChatMessage({
    id,
            userId,
            sender,
            text,
            timestamp,
            loading = false,
}) {
        return {
id,
userId,
sender, // "user" | "assistant"
text,
timestamp,
loading,
        };
        }

// Mapper from backend DTO to frontend model
export function mapDtoToChatMessage(dto) {
    return {
            id: dto.messageId,
            userId: dto.userId,
            sender: dto.role,
            text: dto.content,
            timestamp: dto.timestamp,
  };
}